statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -1112,11 +1112,11 @@ def predict(self, start=None, end=None, dynamic=None, full_results=False,
             if dynamic < 0:
                 raise ValueError('Dynamic prediction cannot begin prior to the'
                                  ' first observation in the sample.')
-            if dynamic > end:
+            elif dynamic > end:
```

Comment:
Matching Stata, I let dynamic be an int, date string (if we have dates), or a bool. bool says to use dynamic prediction for the whole sample, and int or date string allows starting dynamic prediction at a specific point in the sample. The bool behavior matches the behavior in `tsa.arima`, etc.